walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,19 +15,137 @@
+    for(Event event : events){
+      TimeRange eventTime = event.getWhen();
+      
+      /* Checks if the {@code event} has any attendees that are also an attendee for the new meeting {@code request}
+       * If the two collections are disjoint, skip this event
+       */
+      if(checkIfEventsAreDisjoint(event, request, hasOptionalAttendees)){
+          continue;
+      }
+      // List of all times that the meeting can take place
+      ArrayList<TimeRange> addTimes = new ArrayList<TimeRange>();
+      // List of all the times the meeting cannot take place
+      ArrayList<TimeRange> removeTimes = new ArrayList<TimeRange>();
+
+      // Checks if the {@code eventTime} overlaps with the current {@code time}
+      for(TimeRange time: availableMeetingTimes){
+          if(time.overlaps(eventTime)) {
+            timesAroundEvent(addTimes, removeTimes, request.getDuration(), time, eventTime);
+          }
+      }
+      availableMeetingTimes.removeAll(removeTimes);
+      availableMeetingTimes.addAll(addTimes);
+      }
+      return sortArray(availableMeetingTimes);
+    }
+  
+  /*
+   * Resolves conflicting event times by checking before and after the {@code eventTime} for
+   * available times with appropriate {@code newMeetingDuration}
+   *
+   * @param addTimes: list of all the available time ranges for the new meeting
+   * @param removeTimes: list of all the time ranges where the meeting cannot be held
+   * @param newMeetingDuration: the duration of requested meeting
+   * @param time: The conflicting time that the function is checking around
+   * @param eventTime: the time of the event 
+   */ 
+  private void timesAroundEvent(ArrayList<TimeRange> addTimes, ArrayList<TimeRange> removeTimes, long newMeetingDuration, TimeRange time, TimeRange eventTime) {
+    // Create a Time Range for before the {@code eventTime}
+    TimeRange beforeConflict = TimeRange.fromStartEnd(time.start(), eventTime.start(), false);
+    // Create a Time Range for after teh {@code eventTime}
+    TimeRange afterConflict = TimeRange.fromStartEnd(eventTime.end(), time.end(), false);
+
+    // Checks if the new Time Ranges created are longer than the {@code newMeetingDuration}
+    if(beforeConflict.duration() >= newMeetingDuration){
+      addTimes.add(beforeConflict);
+    }
+    if(afterConflict.duration() >= newMeetingDuration){
+      addTimes.add(afterConflict);
+    }
+    removeTimes.add(time);
+  }
+
+  /*
+   * Returns a sorted array in chronological order
+   */
+  private ArrayList<TimeRange> sortArray(Collection<TimeRange> arr){
+    ArrayList<TimeRange> sortedArr = new ArrayList<>(arr);
+    Collections.sort(sortedArr, new Comparator<TimeRange>(){
+      @Override
+      public int compare(TimeRange t1, TimeRange t2){
+        return Long.compare(t1.start(), t2.start());
+      }
+    });
+    return sortedArr;
+  }
 
+  /* Checks if the {@code event} and the {@code request} are disjointed in their attendees and optional attendees
```

Comment:
Clarify that the optional attendee checking only happens when the last parameter is set to `true`.